core/src/main/scala-2.13/org/apache/spark/util/TimeStampedHashMap.scala

```diff
@@ -0,0 +1,143 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.util
+
+import java.util.Map.Entry
+import java.util.Set
+import java.util.concurrent.ConcurrentHashMap
+
+import scala.collection.JavaConverters._
+import scala.collection.mutable
+
+import org.apache.spark.internal.Logging
+
+private[spark] case class TimeStampedValue[V](value: V, timestamp: Long)
```

Comment:
In cases like this, it may be possible to have a common superclass of both 2.12 / 2.13 implementations that holds much of what is common to both. For small classes, it may not be worth the overhead. Here, I don't know? any opinions?

Generally speaking, the only differences in the implementations of mutator methods like ++ and -- changing to updated, removed.